adagrams/game.py

```diff
@@ -1,11 +1,157 @@
+        if letter in letter_dict.keys():
+            letter_dict[letter] += 1
+        else:
+            letter_dict[letter] = 1
+    
+    # iterate through each letter in the given word.
+    # returns True if all letters are in the letter_bank within correct quantity
+    # returns False otherwise 
+
+    for letter in word:
+        letter = letter.upper()
+        if letter not in letter_bank or letter_dict[letter] == 0:
+            return False
+        else:
+            letter_dict[letter] -= 1
+    return True
 
 def score_word(word):
-    pass
+    """
+    Returns the score of a given word as defined by the Adagrams game.
+    """
+    value_dic = {
+        'A': 1, 'E': 1, 'I': 1, 'O': 1, 'U': 1, 'L': 1, 'N': 1, 'R': 1, 'S': 1, 'T': 1,
+        'D': 2, 'G': 2,
+        'B': 3, 'C': 3, 'M': 3, 'P': 3,
+        'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4,
+        'K': 5,
+        'J': 8, 'X': 8,
+        'Q': 10, 'Z': 10
+    }
+
+    # Initialize a variable called sum at 0
+    sum = 0
+
+    # Iterate over each letter in the given word and add its corresponding score to the sum
+    for letter in word:
+        sum += value_dic[letter.upper()]
+    
+    # If the word length is 7, 8, 9 or 10, add an addtional 8 score to the sum
+    if 7 <= len(word) <= 10:
+        sum += 8
+    return sum
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    """
+    Looks at the list of word_list and calculates which of these words has 
+    the highest score, applies any tie-breaking logic, 
+    and returns the winning word in a tuple.
+    """
+
+    # initializing a variable "highest_score" to track highest score 
+    # and a list "highest_scoring_words" to track highest_scoring words.
+
+    highest_score = 0
+    highest_scoring_words = []
+
+    # iterate over each word inside word_list to 
+    # compute that word's score and update the highest_score variable
+    # and highest_scoring_words list 
+    for word in word_list:
+        score = score_word(word)
+        if score > highest_score:
+            highest_scoring_words.clear()
+            highest_score = score
+            highest_scoring_words.append(word)
+        elif score == highest_score:
+            highest_scoring_words.append(word)
+    
+    # initialize the variables "shortest_word_length" and "shortest_word" 
+    # to track highest-scoring word with shortest length for applying 
+    # tie-breaker rules (Winning word is the first spotted 10-letter word. 
+    # If there is no 10 letter word, the shortest word wins.)
+
+    shortest_word_len = len(highest_scoring_words[0])
+    shortest_word = highest_scoring_words[0]
+    for word in highest_scoring_words:
+        if (len(word) == 10):
+            return (word, score_word(word))
+        elif len(word) < shortest_word_len:
+            shortest_word_len = len(word)
+            shortest_word = word
+    return (shortest_word, score_word(shortest_word))    
```

Comment:
Nice job with your `get_highest_word_score()` function! ✨ 

(optional) Consider how you could refactor this code to only use one loop. Worst case scenario, every word in the list has the same score and length, which would mean the function would have to iterate over the same list twice (`O(2n)`). Think about how you can find the highest scored word while only iterating over the list once. 🤔 💭 ☺️